### PR TITLE
tests/upgrade: copy shared_lib.sh to the test vm

### PR DESCRIPTION
**Pull request by jrusz**
shared_lib.sh now contains some more helper functions and the verification script uses them. Copy it over so that we can source it. Also install all other dependencies before the actual upgrade.

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Comment by jrusz** [comment]
Link to Upgrade test job https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/jobs/3034252818

File: test/data/upgrade8to9/upgrade_verify.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 set -euxo pipefail
 
-source /usr/libexec/tests/osbuild-composer/shared_lib.sh
```
Comment (thozza) [inline test/data/upgrade8to9/upgrade_verify.sh:5]: I'm probably missing some more context, but isn't the osbuild-composer-tests RPM installed on the upgraded system? If not, then this makes sense, if it is, then I'm not sure why is this change needed. Could you please elaborate? Thanks.

Comment (jrusz) [reply test/data/upgrade8to9/upgrade_verify.sh:5]: Just the `osbuild-composer` package from the latest nightly is installed on the test VM. See https://github.com/osbuild/osbuild-composer/blob/main/test/data/upgrade8to9/upgrade_prepare.sh#L11

So I copy the script over to the test VM so it can be sourced. The change is visible in this commit.
